src/cql.rs

```diff
@@ -96,162 +96,114 @@ pub fn process(
-                                filter_string =
-                                    filter_string.replace("{{D1}}", date_str_min.as_str());
-                                // no condition needed, "" stays ""
-
-                                let datetime_str_max = date_range.max.as_str();
-                                let datetime_result_max: Result<DateTime<Utc>, _> =
-                                    datetime_str_max.parse().map_err(|_| {
-                                        FocusError::AstInvalidDateFormat(date_range.max)
-                                    });
-                                let datetime_max = datetime_result_max.unwrap(); // we returned if Err
-                                let date_str_max = format!("@{}", datetime_max.format("%Y-%m-%d"));
-
-                                condition_string =
-                                    condition_string.replace("{{D2}}", date_str_max.as_str());
-                                filter_string =
-                                    filter_string.replace("{{D2}}", date_str_max.as_str());
-                                // no condition needed, "" stays ""
-                            }
-                            ast::ConditionValue::NumRange(num_range) => {
-                                condition_string = condition_string
-                                    .replace("{{D1}}", num_range.min.to_string().as_str());
-                                condition_string = condition_string
-                                    .replace("{{D2}}", num_range.max.to_string().as_str());
-                                filter_string = filter_string
-                                    .replace("{{D1}}", num_range.min.to_string().as_str()); // no condition needed, "" stays ""
-                                filter_string = filter_string
-                                    .replace("{{D2}}", num_range.max.to_string().as_str());
-                                // no condition needed, "" stays ""
-                            }
-                            other => {
-                                return Err(FocusError::AstOperatorValueMismatch(format!("Operator BETWEEN can only be used for numerical and date values, not for {:?}", other)));
-                            }
+            match condition.type_ {
+                ast::ConditionType::Between => {
+                    // both min and max values stated
+                    match condition.value {
+                        ast::ConditionValue::DateRange(date_range) => {
+                            let datetime_str_min = date_range.min.as_str();
+                            let datetime_result_min: Result<DateTime<Utc>, _> = datetime_str_min
+                                .parse()
+                                .map_err(|_| FocusError::AstInvalidDateFormat(date_range.min));
+
+                            let datetime_min = datetime_result_min.unwrap(); // we returned if Err
```

Comment:
```suggestion
                            let datetime_min: DateTime<Utc> = datetime_str_min
                                .parse()
                                .map_err(|_| FocusError::AstInvalidDateFormat(date_range.min))?;
```
We did not return ;)